src/Nancy/Extensions/MemoryStreamExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace Nancy.Extensions
+{
+    using System;
+    using System.IO;
+
+    internal static class MemoryStreamExtensions
+    {
+        public static ArraySegment<byte> GetBufferSegment(this MemoryStream stream)
+        {
+            var buffer = default(ArraySegment<byte>);
+
+#if DOTNET5_4
+            stream.TryGetBuffer(out buffer);

```

Comment:
I assume if TryGetBuffer fails, that .ToArray() would also fail. I don't see much of a way around this.